### PR TITLE
8322503: Shenandoah: Clarify gc state usage

### PR DESCRIPTION
**Pull request by earthling-amzn**
This is a follow up to https://github.com/openjdk/jdk/pull/17112.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322503](https://bugs.openjdk.org/browse/JDK-8322503): Shenandoah: Clarify gc state usage (**Task** - P4)

### Reviewers
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - **Reviewer**) ⚠️ Review applies to [13fe4565](https://git.openjdk.org/jdk/pull/17163/files/13fe45653718473c2c7c97ba3880ecd881b04fd5)
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17163/head:pull/17163` \
`$ git checkout pull/17163`

Update a local copy of the PR: \
`$ git checkout pull/17163` \
`$ git pull https://git.openjdk.org/jdk.git pull/17163/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17163`

View PR using the GUI difftool: \
`$ git pr show -t 17163`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17163.diff">https://git.openjdk.org/jdk/pull/17163.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17163#issuecomment-1863517500)

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp [hunk 1]
```diff
@@ -293,12 +293,12 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
...
 
-  void set_gc_state_mask(uint mask, bool value);
+  void set_gc_state(uint mask, bool value);
 
 public:
   char gc_state() const;
-  void set_gc_state_all_threads();
-  bool has_gc_state_changed() { return _gc_state_changed; }
+  void propagate_gc_state_to_java_threads();
+  bool has_gc_state_changed() const { return _gc_state_changed; }
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp:301]: The name changes per @shipilev look good. If we have a sentence of documentation in the header file for each of these methods, that makes things even clearer (and is not subject to limitations on the length of the method name :-)

**Review by ysramakrishna — APPROVED** [review]

**Comment by earthling-amzn** [comment]
/integrate

Added comments to new methods and synced up with tip to see if it addresses GHA build failure on macosx.

**Review by lgxbslgx — APPROVED** [review]
Looks good.

**Pull request closed without merging**